Add partial check-in/check-out indicators and screen reader accessibility features to datepicker – indicate by other than color (WooExt-330)

m-muhsin opened the pull request:
### All Submissions:

<!-- Mark completed items with an [x] -->

* [x] Does your code follow the [WooCommerce Sniffs](https://github.com/woocommerce/woocommerce-sniffs/) variant of WordPress coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?
* [ ] Will this change require new documentation or changes to existing documentation?

---

### Changes proposed in this Pull Request:

This PR enhances the accessibility and visual clarity of the accommodation booking datepicker by making Accessibility Improvements and Visual Enhancements.

Twenty Twenty-Five:
<img width="667" height="618" alt="image" src="https://github.com/user-attachments/assets/556e46f7-c58b-4155-9e52-6b7d6d87a108" />

Storefront:
<img width="445" height="436" alt="image" src="https://github.com/user-attachments/assets/e38648b0-c9ef-408f-bf9c-0663f57c0bcc" />


WooExt-330.

Closes #553
Closes https://linear.app/a8c/issue/WOOACBK-76/tracking-issue-for-pr-550-add-screen-reader-accessibility-features-to

### Steps to test the changes in this Pull Request:

1. Navigate to a product page with an Accommodation Booking product type
2. Open the booking datepicker calendar
3. Test visual indicators like the icons and tooltips show in the above screenshots.
5. Test accessibility with screen reader
6. Test date selection flow
7. Test with different date availability scenarios

### Changelog entry

> Add - Enhanced datepicker accessibility with screen reader support and improved visual indicators for booking availability status.

Commit:
Add screen reader accessibility features to datepicker

- Introduced visually hidden text for screen readers to enhance accessibility for datepicker cells.
- Implemented functions to format and add accessible text for fully booked and partially available dates.
- Updated styles for fully booked dates to include visual indicators (crosses) for better user experience.
- Ensured that screen reader text is re-added after date selection triggers a refresh.

iamdharmesh commented on src/js/booking-form.js:
```diff
@@ -69,6 +69,74 @@ import {
+	 */
+	const formatDateForScreenReader = ($cell) => {
+		const $dayElement = $cell.find('span, a').first();
+		const day = $dayElement.contents().first().text().trim();
```
How about add formatted date [attribute](https://github.com/woocommerce/woocommerce-accommodation-bookings/blob/162e404a8edf94c47c89f87888bd463bffc6e3df/src/js/booking-form.js#L18-L70) and use that directly instead of parsing content and generate it?

iamdharmesh commented on src/js/booking-form.js:
```diff
@@ -131,6 +222,30 @@ import {
+					const formattedDate = formatDateForScreenReader($cell);
+					const accessibleText = `${formattedDate} ${__('Fully booked and unavailable.', 'woocommerce-accommodation-bookings')}`;
+					addAccessibleText($cell, accessibleText);
+				});
```
This seems duplicate of code we have added above, maybe make a function and re-use it?

iamdharmesh commented on src/css/frontend.scss:
```diff
@@ -24,26 +24,129 @@
+		span, a {
+			position: relative;
+
+			// Add a diagonal slash from top-left to bottom-right
```
How about adding a 2 colour (with 3:1 contrast ratio) as suggested [here](https://github.com/woocommerce/woocommerce-accommodation-bookings/issues/201#issuecomment-506764555)? @dkotter Do you have any context on that partially booked day (why was it removed or why it didn't used etc...)?

dkotter replied:
I don't recall at this point and there's lots of back and forth on that issue. Seemed we ended up with some fixes in https://github.com/woocommerce/woocommerce-accommodation-bookings/pull/303 but not sure why we didn't go with this design approach

dkotter replied:
I will say we need to be careful here to not introduce any regressions as anytime in the past we've made changes around availability, often there scenarios we weren't considering that ended up having problems. I wouldn't be surprised if that's the reason we took the approach we did, though doesn't mean we can't take a different approach now. But don't want us to go too far down a rabbit hole on this one

iamdharmesh replied:
Thanks for the insight here @dkotter. For this PR we are not making any changes around availability logic and its only for the UI updates. So, we are good here. However, I’m mainly concerned about the new UI and ensuring that it’s easy for the end user to understand. As I mentioned earlier, the UI suggested [here](https://github.com/woocommerce/woocommerce-accommodation-bookings/issues/201#issuecomment-506764555) looks good to me, but any input from you or @jeffpaul would definitely be helpful.

iamdharmesh replied:
Finalized [this](https://github.com/woocommerce/woocommerce-accommodation-bookings/pull/550#issuecomment-3663465020) UI

iamdharmesh reviewed (changes requested):
Thanks for the PR @m-muhsin. Added few questions could you please check. Thanks

Commit:
Add `data-day` attribute to date cells and centralize accessible text generation for booked dates.

Commit:
Refine date picker styling for fully booked dates and update selector specificity.

iamdharmesh commented on src/js/booking-form.js:
```diff
@@ -65,10 +65,111 @@ import {
+		$form.find('.fully_booked_start_days').each(function() {
+			const $cell = $(this);
+			const formattedDate = formatDateForScreenReader($cell);
+			const accessibleText = `${formattedDate} ${__('Available for check-out only.', 'woocommerce-accommodation-bookings')}`;
```
I noticed that formattedDate is not getting added to the screen reader text. Could you please check whats wrong with it? same for other as well.

m-muhsin replied:
This is now sorted: 

<img width="1517" height="840" alt="Image" src="https://github.com/user-attachments/assets/9d7d6171-9c94-4a25-8d32-944bf20fc24d" />

Thanks for the catch.

iamdharmesh reviewed (changes requested):
Thanks for the changes @m-muhsin. Added note around formattedDate not getting added. Could you please help to check? Also, Eslint action is getting failed, would be great if you can fix the reported errors there as well. Thanks.

Commit:
Merge branch 'trunk' into fix/wooext-328

Commit:
Introduce shared month and day name constants, enhance screen reader date formatting by robustly retrieving date components, and refine `addAccessibleText` targeting.

Commit:
refactor: improve code readability and consistency with updated formatting.

iamdharmesh reviewed (commented):
Thanks for the changes @m-muhsin. It looks good now. Just waiting for conclude on [this](https://github.com/woocommerce/woocommerce-accommodation-bookings/pull/550#discussion_r2601234209), once done. we are good to go.

Btw, this is fix for "WooExt-330" right? If yes, could you please update PR description. Thanks.

m-muhsin commented:
> Btw, this is fix for "WooExt-330" right? If yes, could you please update PR description. Thanks.

Yes, correct. I have updated accordingly. Thank you.

vikrampm1 commented:
@m-muhsin is there is anything pending here or did you address all the feedback above? If yes, can you re-request review to @iamdharmesh so this can be progressed further?

Commit:
Enhance date picker styling for partially booked dates with improved visual indicators and updated gradients for availability representation.

m-muhsin commented:
Hi @vikrampm1. We had a discussion around this in yesterday's call. We decided to implement it like this:

<img width="697" height="892" alt="image" src="https://github.com/user-attachments/assets/0efad075-4bf8-4875-883b-24132b797f9b" />

I have done it likewise now.

Thanks!

iamdharmesh commented on src/js/booking-form.js:
```diff
@@ -65,10 +65,155 @@ import {
+		);
+		return `${MONTH_NAMES[date.getMonth()]} ${day}, ${dataYear}, ${
+			DAY_NAMES[date.getDay()]
+		},`;
```
```suggestion
		return `${MONTH_NAMES[date.getMonth()]}, ${dataYear}, ${
			DAY_NAMES[date.getDay()]
		},`;
```

Should add month name, year and day name only. As suggested in recommended fix, Adding date here make voiceover announce it twice.

vikrampm1 commented:
@m-muhsin it would be great to get these addressed soon so we can move this to the next step. Thanks!

m-muhsin committed:
Remove repeated date.

Co-authored-by: Dharmesh Patel <dspatel44@gmail.com>

m-muhsin commented:
@iamdharmesh I have now incorporated your feedback. Thanks!

CC: @vikrampm1

iamdharmesh commented:
Thanks for the changes @m-muhsin. Changes looks good. However, I noticed that accessibility text not getting added on months other than current. Example: It works fine for current month (Dec) but not for the January

iamdharmesh commented:
@m-muhsin Reminder for addressing this.

Commit:
Merge branch 'trunk' into fix/wooext-328

m-muhsin commented:
Thanks for the reminder, @iamdharmesh. I was trying to replicate it but could not. I even checked for 2027 just in case it was for the following year.

<img width="679" height="865" alt="image" src="https://github.com/user-attachments/assets/b104fa0e-3bce-4cbe-afa5-e42d1e45db06" />

iamdharmesh commented:
**Note:** As discussed over call UI part is work well but accessibility text not getting added on months other than current. Thanks.

Commit:
Enhance booking form attributes by adding data-month and data-year for improved accessibility.

Commit:
Add screen reader support for datepicker navigation in booking form so that it applies to other months than the current month alone.

m-muhsin commented:
@iamdharmesh I have now added even listeners to change of month. So, the last reported issue is now sorted. Thanks!

iamdharmesh commented on src/js/booking-form.js:
```diff
@@ -177,4 +333,38 @@ import {
+		'change',
+		'.ui-datepicker-month, .ui-datepicker-year',
+		function () {
+			setTimeout(() => {
```
Do we need this `setTimeout`? If we don't have proper event to tag this then we might look into fix the issues in bookings itself if that's more solid way to fix this.

m-muhsin replied:
It was required for when testing with a throttled network. I have removed it now. Shall we address it upstream if it comes again?

m-muhsin replied:
@iamdharmesh  as discussed I added the `setTimeout` back along with a comment explaining it.

iamdharmesh commented:
Thanks for fixing it @m-muhsin. Added one question to get some information.

m-muhsin committed:
Remove setTimeout

m-muhsin commented:
@iamdharmesh as mentioned, in [this comment](https://github.com/woocommerce/woocommerce-accommodation-bookings/pull/550#discussion_r2679614261), can we address it in upstream if it comes up again? I removed the setTimeout which is required for slower networks.

m-muhsin committed:
Update booking form to wait for datepicker update before adding screen reader text for improved accessibility.

m-muhsin commented:
@iamdharmesh I felt better to add the setTimeout so did it with a comment explaining it.

iamdharmesh commented on src/css/frontend.scss:
```diff
@@ -24,26 +24,115 @@
+	word-wrap: normal !important;
+}
+
+fieldset.wc-bookings-date-picker {
```
@m-muhsin We have updated markup around `fieldset` in the PR https://github.com/woocommerce/woocommerce-bookings/pull/4098. So, just curious if it is fine to keep using `.wc-bookings-date-picker` as earlier. Could you please update if it's fine. Thank you.

m-muhsin replied:
Thanks for noting. I have removed the `fieldset` from the selector now.

iamdharmesh reviewed (approved):
Thanks for the changes @m-muhsin PR looks good. Added one minor note around markup changes in different ongoing PR. But pre-approving to allow starting QA on this.

m-muhsin committed:
Refactor date picker styling by updating the selector for the bookings date picker to improve CSS specificity.

m-muhsin commented:
I have now also updated the colors for increased contrast:

<img width="639" height="514" alt="image" src="https://github.com/user-attachments/assets/b76d6328-2327-4a12-a7b7-d602ecd54de2" />

iamdharmesh commented:
> I have now also updated the colors for increased contrast:

Thanks @m-muhsin. The black feels bit sharp, is there any problem with current color? is it not meeting contrast expectations?

m-muhsin commented:
@iamdharmesh was aiming for good contrast, really.
<img width="1267" height="452" alt="image" src="https://github.com/user-attachments/assets/da42717e-863c-43ef-8be0-7d79abac7479" />

m-muhsin committed:
style: Update unavailable half color from black to dark gray in frontend datepicker styles.

m-muhsin commented:
Updated the color from #000000 to #666666 as per discussion with Dharmesh.

<img width="1218" height="906" alt="image" src="https://github.com/user-attachments/assets/b4043eaa-ac8b-4f2b-907e-e2c0263ae953" />

Relevant PR: https://github.com/woocommerce/woocommerce-bookings/pull/4125

m-muhsin committed:
feat: Adjust opacity of fully booked dates in the date picker to 1 and refine the selector.

m-muhsin committed:
Refactor: Adjust date picker CSS selectors by adding a new class and simplifying existing ones.

m-muhsin committed:
style: Standardize green background color for bookable and partially available dates.

m-muhsin committed:
refactor: Update datepicker unavailable date background color and set partially available date backgrounds to transparent.

m-muhsin committed:
fix: prevent partial booking gradients from applying to active datepicker days

qasumitbagthariya committed:
Merge branch 'trunk' into fix/wooext-328

qasumitbagthariya reviewed (commented):
**QA Update:** ⚠️
---

@m-muhsin 

Issue 1. Today’s date is 30 Jan, and based on the UI behavior, the checkout date for my booking should be displayed in a fully black color to indicate it is the current date. However, this is not being reflected correctly in the UI.

<img width="1489" height="650" alt="image" src="https://github.com/user-attachments/assets/6571babe-ccd0-4848-867f-81a94fa6f464" />


Issue 2. I have set setting > the Minimum number of nights allowed in a booking to 1 and the Maximum number of nights allowed in a booking to 7. However, as shown in the video, when selecting Feb 11–12, the background color appears differently, while selecting Feb 12–13 shows a fully green background.

The background color behavior should be consistent for both date ranges when they fall within the allowed booking rules.

https://github.com/user-attachments/assets/fda33756-3878-4a62-bb62-99875660c90a

cc @iamdharmesh

m-muhsin commented:
Hi @qasumitbagthariya 

> Issue 2. I have set setting > the Minimum number of nights allowed in a booking to 1 and the Maximum number of nights allowed in a booking to 7.

The right setting for this is: `Availability > Max bookings per block`

And 12 is having a darker shade of green because you are currently on that date, selected.

Will work on a fix for Issue 1.

Thanks!

m-muhsin committed:
feat: Add info icons and tooltips for partially available dates in the booking form

m-muhsin committed:
feat: Dynamically set background color for partially available dates in the booking form

m-muhsin committed:
feat: Implement MutationObserver for dynamic updates of booking form icons and screen reader text

m-muhsin commented:
@iamdharmesh @qasumitbagthariya I have addressed the issues now.

Also, added icons like so:

<img width="676" height="620" alt="image" src="https://github.com/user-attachments/assets/8579a07d-7e62-4b08-8f22-1cec82e53eae" />

m-muhsin committed:
refactor: Optimize MutationObserver logic to prevent redundant updates in booking form

m-muhsin committed:
refactor: Remove dynamic background color setting for partially available dates and simplify related styles

m-muhsin committed:
Merge branch 'trunk' into fix/wooext-328

m-muhsin commented:
As per discussion with @iamdharmesh today, I have now removed the diagonal background color differences to indicate partial availability, now that we are showing icons with tooltips.

Twenty Twemty-Five:
<img width="669" height="624" alt="image" src="https://github.com/user-attachments/assets/5e76e068-338f-44e8-a871-cc1312ecb172" />

Storefront: 
<img width="444" height="450" alt="image" src="https://github.com/user-attachments/assets/b4c6fe42-d8c7-47ba-8946-4ca76d8c9d56" />

Thanks!

iamdharmesh commented on src/css/frontend.scss:
```diff
@@ -24,26 +24,154 @@
 	.ui-datepicker td.fully_booked {
 		span, a {
-			background-color: #c0392b !important;
+			background-color: #575757 !important;
```
As we are dropping idea of playing with color how about, skip the updating color here and keep it as it is?

m-muhsin replied:
Then it would look like this on Storefront. Is it okay?

<img width="418" height="425" alt="Image" src="https://github.com/user-attachments/assets/417d5c64-46f3-4ee9-bc1c-c4bf2b279f44" />

m-muhsin replied:
And like this after removing `opacity: 1;`

<img width="421" height="439" alt="image" src="https://github.com/user-attachments/assets/5d3da5a8-9fc0-4d42-ad19-adbf105c731c" />

iamdharmesh reviewed (commented):
Thanks for the changes @m-muhsin. This looks good now.

Few last things and this will be ready for the QA:
- ESLint check is failing
- Copilot has some minor feedback, could you please check?
- PR description seems very old, could you please update it to the latest along with test steps?

m-muhsin committed:
Revert color changes on fully booked date background

m-muhsin committed:
refactor: Remove styling for fully booked dates in date picker

m-muhsin committed:
style: Show tooltip on hover for partial availability and remove titles for fully booked dates

m-muhsin committed:
style: Improve SCSS formatting and fix linting issues

m-muhsin committed:
chore: Add browser environment to ESLint configuration

m-muhsin committed:
Address GitHub Gopilot's review feedback.

m-muhsin committed:
style: Enhance date picker styles for fully booked dates and improve tooltip visibility

iamdharmesh reviewed (approved):
Thanks for the changes @m-muhsin. This looks good now. 

**Approving with a note:** Since this change involves the jQuery date picker, we don’t have a proper way to add markup there. DOM manipulation is used to achieve this, which may not be the best approach, but we currently don’t have a better alternative.

qasumitbagthariya reviewed (approved):
### QA Update  ✅ 


-------------

I have verified this PR in the `fix/wooext-328` branch, which has been fixed and is functioning as intended.

I tested the following on this branch:

**TT5**
<img width="1262" height="820" alt="image" src="https://github.com/user-attachments/assets/336d6a6f-34af-4923-985c-6161a0e78668" />

**Storefront**
<img width="1077" height="735" alt="image" src="https://github.com/user-attachments/assets/4f15ce9f-c085-44e9-803a-70999f9a977b" />


https://github.com/user-attachments/assets/00575a41-586e-4061-8f91-df606b037028



**Testing Environment** 

<Details>

- WordPress: 6.9.4
- Theme: Storefront 4.6.2
- Theme: Twenty Twenty-Five 1.4
- WooCommerce - 10.6.2
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome
- OS: macOS 15.2 
- Branch: fix/wooext-328
</Details>


Steps to Test- As mentioned in the PR description.
Test Results - It is working as expected.
Functional Demo / Screencast - 
Special Notes -  Ready for UAT
Testing Document status:
Cases related to this Issue/PR are added to the Critical Flow Wiki pages: 

- [ ] Yes
- [x] Not Required/Applicable for this PR
------------------------------------------------

builtbytay committed:
Merge branch 'trunk' into fix/wooext-328

qasumitbagthariya commented:
Regression / Smoke Test Report  ✅ 
-------

Tested with Archive File created via "php woorelease.phar build repo_URL" (Composer version 2.8.4, npm version 10.8.2, node version 20.18.1)

Status- Working expected with Plugin Archive/Zip file same as fix specific branch.

**Testing Environment** 

<Details>

- WordPress: 6.9.4
- Theme: Twenty Twenty-Five 1.4
- WooCommerce - 10.7.0
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS 15.2 
- Branch: smoke-testing

</Details>

Next Step- Ready to Merge 🚀

iamdharmesh committed:
Merge branch 'trunk' into fix/wooext-328

iamdharmesh committed:
Add tooltip enhancement as per feedback.

faisal-alvi commented on src/css/frontend.scss:
```diff
@@ -24,31 +25,160 @@
+		display: none;
+		position: absolute;
+		right: 0;
+		top: calc(100% + 4px);
```
It seems that this 4 px gap between the button and the tooltip is the root cause of the hover-continuity failure Amber [flagged](https://linear.app/a8c/issue/WOOA11Y-113/a11y-status-relies-on-color-alone-not-programatically-defined-wooext#comment-3aba6083) (Point 1). When the cursor moves from the button downward toward the tooltip it briefly exits the wrapper's `:hover` zone - this could be the reason why the tooltip disappeared before the cursor reached it, making it impossible to hover over the tooltip content.

We can close the gap without losing the visual arrow spacing. Maybe by using the `top: 100%` and shift the visual space into `padding-top` on the tooltip itself:

```scss
.partial-availability-tooltip {
    top: 100%;
    padding: 10px 10px 6px; // visual gap via padding

    &::before {
        bottom: calc(100% - 4px); // adjust arrow offset to compensate
    }
}
```
Please let me know if this works in your testing or we are good with the current approach.

iamdharmesh replied:
I have added CSS to improve :hover zone.

faisal-alvi commented on src/js/booking-form.js:
```diff
@@ -54,21 +54,335 @@ import {
+			}
+			e.preventDefault();
+			e.stopPropagation();
+			const $w = $(this).closest('.partial-availability-info-wrapper');
+			$w.toggleClass(PARTIAL_TOOLTIP_DISMISSED_CLASS);
+			syncPartialAvailabilityTooltipAria($w);
```
Not tested but looking here - it seems like the `toggleClass` call is inverted. When the button receives focus, CSS `:focus-within` already makes the tooltip visible. Pressing Space/Enter hits `toggleClass`, which adds the dismissed class on the very first press - which will immediately hide the tooltip that `:focus-within` just showed. The first keypress would always close, never opens.

If that is true, we should replace `toggleClass` with an explicit check:

```js
const $w = $(this).closest('.partial-availability-info-wrapper');
if ($w.hasClass(PARTIAL_TOOLTIP_DISMISSED_CLASS)) {
    $w.removeClass(PARTIAL_TOOLTIP_DISMISSED_CLASS);
} else {
    $w.addClass(PARTIAL_TOOLTIP_DISMISSED_CLASS);
}
syncPartialAvailabilityTooltipAria($w);
```

iamdharmesh replied:
I think that is expected. Even with the suggested code it will remain the same.

faisal-alvi replied:
Thanks for clarifying.

faisal-alvi commented on src/js/booking-form.js:
```diff
@@ -54,21 +54,335 @@ import {
+	 * @param {jQuery} $form - The booking form element.
+	 */
+	const addPartialAvailabilityIcons = ($form) => {
+		$form.find('.partial-availability-info-wrapper').remove();
```
Every call to `addPartialAvailabilityIcons` starts by destroying all existing icon elements. If a keyboard user is focused on one of those info buttons when this runs (e.g. after a date selection), their focus is silently dropped to `<body>`.

The `appendIcon` condition/guard at line 138 already prevents duplicates. So maybe we can remove this `.remove()` and update existing icons in place instead:

```js
const appendIcon = ( $cell, message ) => {
    const $existing = $cell.find( '.partial-availability-info-wrapper' );
    if ( $existing.length ) {
        // Update text in place - this will preserve the keyboard focus
        $existing.find( '.partial-availability-info' ).attr( 'aria-label', message );
        $existing.find( '.partial-availability-tooltip' ).text( message );
        return;
    }
    // ... rest unchanged
};
```
thoughts?

iamdharmesh replied:
> If a keyboard user is focused on one of those info buttons when this runs (e.g. after a date selection), their focus is silently dropped to <body>.

Mostly this info buttons re added only when calendar get refreshed/reloaded. So, I think focus will be dropped if we skip remove as well. what do you think?

faisal-alvi replied:
That's a fair point. Happy to leave it as is.

faisal-alvi commented on src/js/booking-form.js:
```diff
@@ -54,21 +54,335 @@ import {
-				attributes.title = __(
+				attributes.title = '';
+			} else if (attributes.class.indexOf('fully_booked_end_days') > -1) {
+				attributes.title = '';
```
Nit: Both conditions produce identical output, we can collapse them to one:

```js
if (
    attributes.class.indexOf('fully_booked_start_days') > -1 ||
    attributes.class.indexOf('fully_booked_end_days') > -1
) {
    attributes.title = '';
}
```

faisal-alvi commented on src/css/frontend.scss:
```diff
@@ -24,31 +25,160 @@
+	position: absolute;
+	width: 1px;
+	word-wrap: normal !important;
+}
```
Nit/suggestion: `.screen-reader-text` is an unscoped global selector. WordPress core, Twenty* themes, and many plugins define the same class — a specificity or load-order conflict will silently break either the plugin's visually-hidden styles or the theme's.

We can scope it to prevent collisions, for example:

```scss
.wc-bookings-date-picker-accommodation-booking .screen-reader-text { ... }
```

faisal-alvi commented on src/js/booking-form.js:
```diff
@@ -54,21 +54,335 @@ import {
+			const $monthEl = $datepicker.find('.ui-datepicker-month');
+			const $yearEl = $datepicker.find('.ui-datepicker-year');
+
+			dataMonth = $monthEl.is('select')
+				? $monthEl.val()
+				: MONTH_NAMES.findIndex((m) =>
+						$monthEl.text().toLowerCase().includes(m.toLowerCase())
+				  );
```
Nit but this fallback resolves the month by comparing the datepicker header's text against the English `MONTH_NAMES` array via `findIndex`. On non-English installs the comparison would fail and return `-1`, so `formatDateForScreenReader` would silently return `''` - screen reader text for those cells would be empty. What do you think?

Since `data-month` is always stamped onto every cell by the filter above at line 65, I think we can remove this fallback, or at minimum should have a comment noting it only works in English. thoughts?

iamdharmesh replied:
strings in MONTH_NAMES are translatable. So, I think it worth keeping it and as we can translate those strings I think that is not english only now. what do you think?

faisal-alvi replied:
Good point - the strings are translatable via `__()` so the `findIndex` comparison would work correctly in translated environments too, I missed that.

faisal-alvi reviewed (commented):
@iamdharmesh thanks for adding more commits to address the recent feedback. I reviewed the entire PR and left feedback comments below (One 4px concern and other nits/suggestions) that may be worth considering if you think they make sense. Otherwise, the overall PR looks good.

iamdharmesh committed:
Enhance tooltip accessibility and adjust screen reader text styles in frontend.scss.

faisal-alvi reviewed (approved):
@iamdharmesh 🚀

qasumitbagthariya reviewed (approved):
QA Update: ✅
---

I have verified this PR in the `fix/wooext-328` branch, which has been fixed and is functioning as intended.


1. Ensure that when the tooltip is visible on hover it can be fully hovered with the mouse without disappearing (currently it disappears in the mouse moves over the parts of the tool time that extend beyond the cell boundary.

2. Ensure the tooltip can be dismissed with the escape key. This should be possible for both the tool tip on hover and the tooltip that is visible when the button receives focus.

3. Add aria-expanded and toggle the value from true and false to match the tooltip visibility.

4. When focused, this button should be able to show or hide the tooltip using both the spacebar and return/enter keys.


<img width="1343" height="755" alt="image" src="https://github.com/user-attachments/assets/61ab18aa-5b13-4d53-8f1e-a53d7b3a3b7d" />

**Testing Environment** 

<Details>

- WordPress: 6.9.4
- Theme: Storefront 4.6.2
- Theme: Twenty Twenty-Five 1.4
- WooCommerce - 10.7.0
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome
- OS: macOS 15.2 
- Branch: fix/wooext-328

</Details>

iamdharmesh committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-accommodation-bookings into fix/wooext-328

iamdharmesh committed:
E2E: Update tests.

iamdharmesh committed:
Merge branch 'trunk' into fix/wooext-328
